simpleclient_dropwizard/src/main/java/io/prometheus/client/dropwizard/DropwizardExports.java

```diff
@@ -91,61 +93,71 @@ private static String getHelpMessage(String metricName, Metric metric) {
      */
-    List<MetricFamilySamples> fromTimer(String dropwizardName, Timer timer) {
+    MetricFamilySamples fromTimer(String dropwizardName, Timer timer) {
         return fromSnapshotAndCount(dropwizardName, timer.getSnapshot(), timer.getCount(),
                 1.0D / TimeUnit.SECONDS.toNanos(1L), getHelpMessage(dropwizardName, timer));
     }
 
     /**
      * Export a Meter as as prometheus COUNTER.
      */
-    List<MetricFamilySamples> fromMeter(String dropwizardName, Meter meter) {
+    MetricFamilySamples fromMeter(String dropwizardName, Meter meter) {
         final MetricFamilySamples.Sample sample = sampleBuilder.createSample(dropwizardName, "_total",
                 new ArrayList<String>(),
                 new ArrayList<String>(),
                 meter.getCount());
-        return Arrays.asList(
-                new MetricFamilySamples(sample.name, Type.COUNTER, getHelpMessage(dropwizardName, meter),
-                        Arrays.asList(sample))
-
-        );
+        return new MetricFamilySamples(sample.name, Type.COUNTER, getHelpMessage(dropwizardName, meter),
+                        Arrays.asList(sample));
     }
 
     @Override
     public List<MetricFamilySamples> collect() {
-        ArrayList<MetricFamilySamples> mfSamples = new ArrayList<MetricFamilySamples>();
+        Map<String, MetricFamilySamples> mfSamplesMap = new HashMap<String, MetricFamilySamples>();
+
         for (SortedMap.Entry<String, Gauge> entry : registry.getGauges().entrySet()) {
-            mfSamples.addAll(fromGauge(entry.getKey(), entry.getValue()));
+            addToMap(mfSamplesMap, fromGauge(entry.getKey(), entry.getValue()));
         }
         for (SortedMap.Entry<String, Counter> entry : registry.getCounters().entrySet()) {
-            mfSamples.addAll(fromCounter(entry.getKey(), entry.getValue()));
+            addToMap(mfSamplesMap, fromCounter(entry.getKey(), entry.getValue()));
         }
         for (SortedMap.Entry<String, Histogram> entry : registry.getHistograms().entrySet()) {
-            mfSamples.addAll(fromHistogram(entry.getKey(), entry.getValue()));
+            addToMap(mfSamplesMap, fromHistogram(entry.getKey(), entry.getValue()));
         }
         for (SortedMap.Entry<String, Timer> entry : registry.getTimers().entrySet()) {
-            mfSamples.addAll(fromTimer(entry.getKey(), entry.getValue()));
+            addToMap(mfSamplesMap, fromTimer(entry.getKey(), entry.getValue()));
         }
         for (SortedMap.Entry<String, Meter> entry : registry.getMeters().entrySet()) {
-            mfSamples.addAll(fromMeter(entry.getKey(), entry.getValue()));
+            addToMap(mfSamplesMap, fromMeter(entry.getKey(), entry.getValue()));
+        }
+        return new ArrayList<MetricFamilySamples>(mfSamplesMap.values());
+    }
+
+    private void addToMap(Map<String, MetricFamilySamples> mfSamplesMap, MetricFamilySamples newMfSamples)
+    {
+        if (newMfSamples != null) {
```

Comment:
Why would this be null?

Reply:
`fromGauge` can return null now. I've changed `from*` methods to return a `MetricFamilySample` directly instead of a `List` with only one element.